### PR TITLE
Improve `SqlTypeName` to support more types and also improve error handling

### PR DESCRIPTION
**Pull request by andygrove**
- Improve `SqlTypeName::from_string()` to support more types, using the sql parser to parse parameterized types such as `VARCHAR(n)` and `DECIMAL(p, s)`
- Replaced some `todo!` and `unimplemented!` with `Err` and made corresponding changes in call sites

### CODE REVIEW

File: dask_planner/src/sql/types.rs [hunk 1]
```diff
@@ -269,58 +270,85 @@ impl SqlTypeName {
...
+            "INTERVAL_YEAR_MONTH" => Ok(SqlTypeName::INTERVAL_YEAR_MONTH),
+            "MAP" => Ok(SqlTypeName::MAP),
+            "MULTISET" => Ok(SqlTypeName::MULTISET),
+            "OTHER" => Ok(SqlTypeName::OTHER),
+            "ROW" => Ok(SqlTypeName::ROW),
+            "SARG" => Ok(SqlTypeName::SARG),
+            "BINARY" => Ok(SqlTypeName::BINARY),
+            "VARBINARY" => Ok(SqlTypeName::VARBINARY),
+            "CHAR" => Ok(SqlTypeName::CHAR),
+            "VARCHAR" | "STRING" => Ok(SqlTypeName::VARCHAR),
```
Comment (randerzander) [inline dask_planner/src/sql/types.rs:318]: I've just tried these changes with my own tests.

I can now use "string" types, but varchars w/ a defined limit still fail:
```
Traceback (most recent call last):               
  File "<stdin>", line 1, in <module>                                                                            
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/context.py", line 238, in create_table
    dc = InputUtil.to_dc(                                                                                                   
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/convert.py", line 68, in to_dc
    table = filled_get_dask_dataframe(input_item)                                                           
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/convert.py", line 57, in <lambda>
    filled_get_dask_dataframe = lambda *args: cls._get_dask_dataframe(                                                      
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/convert.py", line 90, in _get_dask_dataframe
    return plugin.to_dc(                                                                               
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/hive.py", line 69, in to_dc
    column_information = {                                                                                    
  File "/opt/conda/envs/rapids/lib/python3.9/site-packages/dask_sql/input_utils/hive.py", line 70, in <dictcomp>
    col: sql_to_python_type(SqlTypeName.fromString(col_type.upper()))                                      
RuntimeError: Internal("Cannot determine SQL type name for 'VARCHAR(65535)'")
```

Comment (andygrove) [reply dask_planner/src/sql/types.rs:325]: I have now pushed changes to support `VARCHAR(n)` and some other parameterized types

**Comment by andygrove** [comment]
`cargo test` is failing with:

```
  /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/pyo3-0.17.1/src/types/any.rs:813: undefined reference to `PyObject_Str'
          collect2: error: ld returned 1 exit status
```

**Comment by andygrove** [comment]
`cargo test` fails on this PR but not on others, so something in this PR is causing a linker issue. Very odd. I am looking into it.

**Comment by andygrove** [comment]
TIL that you cannot reference PyMethods from Rust unit tests (at least, not without additional environment changes). Tests should now be passing.

**Review by ayushdg — APPROVED** [review]

**Pull request merged**